### PR TITLE
Graphical Lasso

### PR DESCRIPTION
**Pull request by weifang1**
Add a new algorithm Graphical Lasso and corresponding test script. Estimated precision matrix is the same as the output of the python script.

### CODE REVIEW

**Comment by blengerich** [comment]
Can one of the admins verify this patch?

File: src/Graph/GraphicalLasso.cpp [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+}
+
+MatrixXf GraphicalLasso::matrixRaiseToHalf(MatrixXf& matrix) {
+    JacobiSVD<MatrixXf> svd(matrix, ComputeThinU | ComputeThinV);
+    MatrixXf D = svd.singularValues().array().sqrt().matrix().asDiagonal();
+    MatrixXf U = svd.matrixU();
+    return U * D * U.transpose();
+}
+
+MatrixXf GraphicalLasso::pseudoInverse(MatrixXf& matrix) {
```
Comment (HaohanWang) [inline src/Graph/GraphicalLasso.cpp]: Please put this into Math.cpp (In Mach folder) since this is a general math solution, not specific to Glasso.

Comment (weifang1) [inline src/Graph/GraphicalLasso.cpp]: OK. How about the function "stdNormalize" below on the line 166? This function would normalize the data matrix column by column.

File: src/Graph/GraphicalLasso.cpp [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        model->updateBeta(model->proximal_operator(in, learningRate));
+        residue = model->cost();
+        epoch++;
+    }
+    return model->getBeta();
+}
+
+void GraphicalLasso::stdNormalize(MatrixXf& matrix) {
+    RowVectorXf mean = matrix.colwise().mean();
+    // unbiased estimation
```
Comment (weifang1) [inline src/Graph/GraphicalLasso.cpp]: I should remove this comment. If I use unbiased estimation for variance, the result is different with the output of the python script.

File: src/Graph/GraphicalLasso.cpp (hunk 1 above)
Comment (HaohanWang) [inline src/Graph/GraphicalLasso.cpp]: I think that one is less useful. We could also consider that one, but we need to give it a more descriptive name if we want to put it into Math

File: src/Graph/GraphicalLasso.cpp (hunk 2 above)
Comment (HaohanWang) [inline src/Graph/GraphicalLasso.cpp]: Let's not worry about unbiased estimation, it doesn't make big differences in practice.

File: src/Graph/GraphicalLasso.cpp (hunk 1 above)
Comment (weifang1) [inline src/Graph/GraphicalLasso.cpp]: Updated. Already moved pseudoInverse to Math.

**Pull request merged**